projects/self-creation-portal/src/app/services/common-service/common.service.ts

```diff
@@ -56,23 +56,33 @@ export class CommonService {
     const queryParams = this.route.snapshot.queryParams;
     const currentPath = this.router.url.split('?')[0];
     const parentRoutes = this.route.parent?.snapshot.url.map(segment => segment.path).join('/') || '';
-
+  
     const sidenavPaths = [
       `${parentRoutes}/${ROUTE_PATHS.SIDENAV.DRAFTS}`,
       `${parentRoutes}/${ROUTE_PATHS.SIDENAV.UP_FOR_REVIEW}`,
       `${parentRoutes}/${ROUTE_PATHS.SIDENAV.SUBMITTED_FOR_REVIEW}`
     ];
     const targetPaths = [
-      `${parentRoutes}/${ROUTE_PATHS.PROJECT_ROUTES.PROJECT_DETAILS}`
+        `${parentRoutes}/${ROUTE_PATHS.PROJECT_ROUTES.PROJECT_DETAILS}`
     ];
+    const createNewPath = `${parentRoutes}/${ROUTE_PATHS.CREATE_NEW.CREATE_NEW}`;
+    
     const issidenavPaths = sidenavPaths.some(tabPath => currentPath.startsWith(tabPath));
     const istargetpaths = targetPaths.some(target => currentPath.startsWith(target))
- 
+    const isCreateNew = currentPath.endsWith(createNewPath);
+  
     if ((istargetpaths && issidenavPaths && Object.keys(queryParams).length > 0) || 
-      (!istargetpaths && Object.keys(queryParams).length > 0)) {
-      this.router.navigate([], {
+    (!istargetpaths && Object.keys(queryParams).length > 0)) {
+      this.router.navigate([], {  
         relativeTo: this.route,
         queryParams: {}
+      }).then(() => {
+      // After clearing query params, navigate to "create-new" if applicable
+      if (isCreateNew) {
+        setTimeout(() => {
+          this.router.navigate([createNewPath], { relativeTo: this.route });
```

Comment:
why setTimeout for route change ? please explain the use case of setTimrout here